packages/clay-multi-select/src/htmlEscape.js

```diff
@@ -0,0 +1,13 @@
+const ampersandRegExp = /&/g;
+const lessThanRegExp = /</g;
+const greaterThanRegExp = />/g;
+const doubleQuoteRegExp = /"/g;
+const singleQuoteRegExp = /'/g;
+
+export const htmlEscape = string =>
+	string
+		.replace(ampersandRegExp, '&amp;')
+		.replace(lessThanRegExp, '&lt;')
+		.replace(greaterThanRegExp, '&gt;')
+		.replace(doubleQuoteRegExp, '&quot;')
+		.replace(singleQuoteRegExp, '&#39;');
```

Comment:
@matuzalemsteles @jbalsas 

For escaping data between `<tag>` and `</tag>`, you need to also escape the forward slash `/`.

Reply:
@samuelkong: How would a `/` be problematic in that position?

Reply:
@wincent
This gets into very obscure and mostly unused parts of the HTML specs. 
`<div/data/` is actually interpreted as `<div>data</div>`. So an unescaped forward slash can be used to close the `<div>` tag.

Reference: https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#rule-1---html-escape-before-inserting-untrusted-data-into-html-element-content

Reply:
Thanks for the explanation @samuelkong, seeing as the cheatsheet doesn't really explain why (it just says, vaguely, "Forward slash is included as it helps end an HTML entity", which is pretty unsatisfying).

Although, seeing as we're escaping `<`, we'd never actually insert `<div/data/` data into the DOM but rather `&lt;/data/`. Would that still be a problem? At first glance, it doesn't seem it would.

Reply:
@wincent
Off the top of my head, I can't think of anything. But we still have to implement the escape for forward slash. If not, you can be the one that goes to marketing and tell them to remove everything about following OWASP guidelines from our marketing material.

Reply:
Oh, absolutely, there's no harm "over"-escaping; I'm just trying to *understand* as opposed to blindly cargo-culting. My googling didn't turn up anything that would explain it.

Reply:
Hey @samuelkong, thanks for chiming in!

I share @wincent's desire for understanding what are the requirements we have to meet for security fixes so we can make sure we don't miss them next time. Do you have some specific links to where we say we follow OWASP guidelines and to the point in those guidelines where what you mention is specified for this case? That would be really formative for us!

I've also filed an internal ticket with the security team (https://issues.liferay.com/browse/PTR-1132) to get some guidance and clarification from them to make sure we don't miss anything.

Thanks!

Reply:
@jbalsas 
I don't have anything off the top of my head, but a quick Google searched returned https://portal.liferay.dev/docs/7-0/deploy/-/knowledge_base/d/liferay-portal-security-overview. I couldn't find anything more recent. However, I know I've seen some recent RFP from sales where we says we following OWASP guidelines. As in this link, I think we usually only say we follow OWASP top ten and OWASP top ten includes protecting against injection vulnerability.

I general, we don't specifically mention that we follow OWASP XSS escaping rules to customers unless the customer ask. I also tell our security testing vendor that our escape methods complies with OWASP XSS Guidelines (specifically the OWASP XSS cheat sheet).

Reply:
Thanks @samuelkong!

I had a brief conversation with @topolik and @csierra which is more or less reflected in https://issues.liferay.com/browse/PTR-1132.

The owasp recommendation can be found at https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#output-encoding-rules-summary. Unfortunately, it does not explain why either, but at least is a valid reference.